src/main/java/Duke.java

```diff
@@ -1,10 +1,271 @@
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+
 public class Duke {
+
+    public static boolean hasSpaceError(String[] splitUserInput, String UserInput) {
+        //Checks whether last character is a space
+        int lastCharacterIndex = UserInput.length() - 1;
+        String lastCharacter = UserInput.substring(lastCharacterIndex);
+        if (lastCharacter.equals(" ")) {
+            return true;
+        }
+        //Checks for multiple spaces between words
+        int userInputLength = splitUserInput.length;
+        for (int i = 0; i < userInputLength; i++) {
+            if (splitUserInput[i].equals("")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static String[] processFileLine (String fileLine) {
+        String[] splitFileLine = fileLine.split(" ");
+        String[] processedFileLine = new String[3];
+        //Sets category
```

Comment:
Perhaps better variable naming to reduce the need for comments. Applies to other parts of your code as well